cob_obstacle_distance/src/cob_obstacle_distance.cpp

```diff
@@ -49,22 +56,34 @@ int main(int argc, char **argv)
 
     ros::Subscriber jointstate_sub = nh.subscribe("joint_states", 1, &DistanceManager::jointstateCb, &sm);
     ros::ServiceServer registration_srv = nh.advertiseService("obstacle_distance/registerPointOfInterest" , &DistanceManager::registerPointOfInterest, &sm);
+    ros::ServiceServer distance_prediction_srv = nh.advertiseService("obstacle_distance/predictDistance" , &DistanceManager::predictDistance, &sm);
 
     ROS_INFO("Starting basic_shapes ...\r\n");
-    fcl::Box b(0.3, 0.3, 0.3);
+    fcl::Sphere s(0.1);
+    fcl::Box b(0.1, 0.1, 0.1); // Take care the nearest point for collision is one of the eight corners!!! This might lead to jittering
+
+    Ptr_IMarkerShape_t sptr_Bvh(new MarkerShape<BVH_RSS_t>(sm.getRootFrame(),
+                                                           "package://schunk_description/meshes/lwa4p_extended/arm_1_collision.stl",
+                                                           -0.35,
+                                                           -0.35,
+                                                           0.8));
+
+
+    Ptr_IMarkerShape_t sptr_Sphere(new MarkerShape<fcl::Sphere>(sm.getRootFrame(), s, 0.35, -0.35, 0.8));
 
-    t_ptr_IMarkerShape sptr_Cube(new MarkerShape<fcl::Box>(b, 0.5, -0.5, 1.0));
-    // t_ptr_IMarkerShape sptr_Sphere(new MarkerShape<fcl::Sphere>(1.0, -1.0, -1.0));
-    // t_ptr_IMarkerShape sptr_Cyl(new MarkerShape<fcl::Cylinder>(-1.0, 1.0, -1.0));
+    //Ptr_IMarkerShape_t sptr_Cube(new MarkerShape<fcl::Box>(b, 0.35, -0.35, 0.8));
+    // Ptr_IMarkerShape_t sptr_Sphere(new MarkerShape<fcl::Sphere>(1.0, -1.0, -1.0));
+    // Ptr_IMarkerShape_t sptr_Cyl(new MarkerShape<fcl::Cylinder>(-1.0, 1.0, -1.0));
 
-    sm.addObstacle(sptr_Cube);
-    // sm.addObstacle(sptr_Sphere);
+    sm.addObstacle("Sphere1", sptr_Sphere);
+    sm.addObstacle("arm_1_collision", sptr_Bvh);
+    // sm.addObstacle(sptr_Cube);
     // sm.addObstacle(sptr_Cyl);
```

Comment:
Implemented callback method registerObstacle to process CollisionObject messages. 

TODO: Topic has to be defined. Currently e.g. "/arm_right/obstacle_distance/registerObstacle"